src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -162,45 +177,22 @@ def _get_bids_offers(market: TwoSidedMarket, filters: Dict):
                 offer.serializable_dict() for offer in offers.values())
         return bids_list, offers_list
 
-    def _get_validated_recommendations(
-            self, recommendations: List[Dict]) -> List[BidOfferMatch.serializable_dict]:
-        """Return a validated list of BidOfferMatch instances."""
-        validated_recommendations = []
+    def _validate_recommendations(
+            self, recommendations: List[Dict]) -> None:
+        """Validate a list of BidOfferMatch instances.
+
+        Check for conditions that should terminate the matching process.
+            - Invalid BidOfferMatch structure
+            - Submitting recommendation to an already finished market
+        Raises
+            MycoValidationException: Condition that is supposed to stop the matching process
+        """
         for recommendation in recommendations:
             if not BidOfferMatch.is_valid_dict(recommendation):
-                raise MycoValidationException("BidOfferMatch is not valid")
+                raise MycoValidationException(f"BidOfferMatch is not valid {recommendation}")
             market = self.markets_mapping.get(recommendation.get("market_id"))
             if market is None:
                 # The market doesn't exist
                 raise MycoValidationException(
-                    f"Market with id {recommendation.get('market_id')} doesn't exist.")
-            if market.readonly:
-                # The market is already finished
-                # TODO: we're clearing markets cache after each market cycle so is this check
-                # TODO: .. really relevant? (All finished markets should be deleted already)
-                raise MycoValidationException(
-                    "Cannot match trades in a finished market.")
-
-            # Get the original bid and offer from the market
-            market_bids = [
-                market.bids.get(bid.get("id")) for bid in recommendation.get("bids")]
-            market_offers = [
-                market.offers.get(offer.get("id")) for offer in recommendation.get("offers")]
-
-            if not (all(market_bids) and all(market_offers)):
-                # Offers or Bids either don't belong to market or were already matched
-                continue
-            try:
-                market.validate_bid_offer_match(
-                    market_bids,
-                    market_offers,
-                    recommendation.get("trade_rate"),
-                    recommendation.get("selected_energy")
-                )
-            except InvalidBidOfferPairException:
-                continue
```

Comment:
These were removed because they are duplicates available in the match_recommendations function of the Two Sided market
These are non blockers, they do not stop the matching process, they just omit the recommendation